src/slice/new_iter.rs

```diff
@@ -0,0 +1,495 @@
+
+	#[inline]
+	fn nth(&mut self, n: usize) -> Option<Self::Item> {
+		if n >= len!(self) {
+			// This iterator is now empty.
+			unsafe {
+				// End can't be 0 if T isn't ZST because ptr isn't 0 and end >= ptr
+				self.ptr = NonNull::new_unchecked(self.end as *mut T);
+				self.bit_ptr = self.bit_end;
+			}
+			return None;
+		}
+		// We are in bounds.
+		//
+		// `n` shows bits, not bytes, thus pointer should be moved by
+		// `n / (size_of::<$store_type>() * 8)`
+		// and bitmask should be rotated by `n & (BITS - 1)`, because
+		// any number over those bits would simply do full circle.
+		// How funny, that `BITS` is `(size_of::<$store_type>() * 8)`!
+		unsafe {
+			self.ptr = NonNull::new_unchecked(
+				self.ptr.as_ptr().add(n / T::BITS as usize),
+			);
+		}
+		self.bit_ptr =
+			O::rot_forward(self.bit_ptr, (n & (T::BITS as usize - 1)) as u32);
+		// Now pointers are set, time to call `next()` without emptyness check.
+		let bit = unsafe {
+			let old = (*self.ptr.as_ptr() & *self.bit_ptr) != T::ZERO;
+
+			if self.bit_ptr == Self::last_bit() {
+				forward_ptr_unchecked!(self);
+			} else {
+				// self.bit_ptr <<= 1;
+				self.bit_ptr = O::rot_forward(self.bit_ptr, 1);
+			}
+			old
+		};
+
+		Some(bit)
+	}
+
+	#[inline]
+	fn last(mut self) -> Option<Self::Item> {
+		self.next_back()
+	}
+}
+
+impl<'a, T: BitMemory, O: BitOrder> DoubleEndedIterator for Iter<'a, T, O> {
+	#[inline]
+	fn next_back(&mut self) -> Option<Self::Item> {
+		if is_empty!(self) {
+			return None;
+		} else {
+			if self.bit_end == Self::start_bit() {
+				unsafe {
+					backward_end_unchecked!(self);
+				}
+			}
+
+			unsafe { Some(next_back_bit_unchecked!(self, *self.end, T)) }
+		}
+	}
+
+	#[inline]
+	fn nth_back(&mut self, n: usize) -> Option<Self::Item> {
+		if n >= len!(self) {
+			// This iterator is now empty.
+			self.end = self.ptr.as_ptr();
+			self.bit_end = self.bit_ptr;
+			return None;
+		}
+		// We are in bounds.
+		//
+		// `n` shows bits, not bytes, thus pointer should be moved by
+		// `n / (size_of::<$store_type>() * 8)`
+		// and bitmask should be rotated by `n & (BITS - 1)`, because
+		// any number over those bits would simply do full circle.
+		// How funny, that `BITS` is `(size_of::<$store_type>() * 8)`!
+		//
+		// SAFETY: `n` is usize, `BITS` >= 8, `usize / 8` is always less
+		// than isize::MAX.
+		self.end =
+			unsafe { self.end.offset(-((n / T::BITS as usize) as isize)) };
+		self.bit_end =
+			O::rot_backward(self.bit_end, (n & (T::BITS as usize - 1)) as u32);
+		// Now pointers are set, time to call `next()` without emptyness check.
+		let bit = unsafe {
+			if self.bit_end == Self::start_bit() {
+				backward_end_unchecked!(self);
+			}
+			next_back_bit_unchecked!(self, *self.end, T)
+
+			// (*self.end & self.bit_end) != 0
+		};
+
+		Some(bit)
+	}
+}
+
+impl<T: BitMemory, O: BitOrder> FusedIterator for Iter<'_, T, O> {}
+
+/*
+/// Builder function for new iter.
+pub fn new_iter<'a, T: BitStore + BitMemory, O: BitOrder>(
```

Comment:
Need help with this thing, @myrrlyn 
I think, I'm doing something wrong, because this just doesn't work.
I can't write tests before there's a way to create this iterator and this is bad.